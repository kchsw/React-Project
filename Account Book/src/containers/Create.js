import React, { Component } from 'react';
import CategorySelect from '../components/CategorySelect'
import PriceForm from '../components/PriceForm'
import { Tabs, Tab } from '../components/Tabs'
import { TYPE_INCOME, TYPE_OUTCOME } from '../utility'
import Ionicon from 'react-ionicons'
import WithContext from '../WithContext'
import { withRouter } from 'react-router-dom'


const tabsText = [ TYPE_INCOME, TYPE_OUTCOME ]

class Create extends Component {
    constructor(props) {
        super(props);
        const { items, categories } = props.data
        const { id } = props.match.params
        this.state = {  
          tabView: (id && items[id]) ? categories[items[id].cid].type : TYPE_INCOME,
          selectedCategory: (id && items[id]) ? categories[items[id].cid] : null
        }
    }
    componentDidMount(){
      const { id } = this.props.match.params
      this.props.actions.getEditData(id).then(data => {
        const { categories, editItem } = data
        this.setState({  
          tabView: (id && editItem) ? categories[editItem.cid].type : TYPE_INCOME,
          selectedCategory: (id && editItem) ? categories[editItem.cid] : null
        })
      })
    }
    changeView = (index) => {
        this.setState({
            tabView: tabsText[index]
        })
    }
    cancelSubmit = () => {
      this.props.history.push('/')
    }
    submitForm = (data, editMode) => {
      if(!editMode){
        this.props.actions.createItem(data, this.state.selectedCategory.id)
      }else{
        this.props.actions.updateItem(data, this.state.selectedCategory.id)
      }
      this.props.history.push('/')
    }
    selectCategory = (category) => {
      this.setState({
        selectedCategory: category
      })
    }
    render() { 
        const { match, data } = this.props
        const { items, categories } = data
        const { id } = this.props.match.params
        const editItem = (id && items[id]) ? items[id] : {}
        const tabIndex = tabsText.findIndex(text => text === this.state.tabView)
        const filterCategories = Object.values(categories).filter(category => category.type === this.state.tabView)
        return (  
            <div className="card" style={{width: '65%', margin: '0 auto'}}>
                <div className="card-body">
                    <div className="content-area">
                        <Tabs activeIndex={tabIndex} onTabChange={this.changeView}>
                          <Tab>
                            <Ionicon
                              className="rounded-circle mr-2"
                              fontSize="25px"
                              color={'#007bff'}
                              icon="ios-add-circle"
                            />
                            收入
                          </Tab>
                          <Tab>
                            <Ionicon
                              className="rounded-circle mr-2"
                              fontSize="25px"
                              color={'#007bff'}
                              icon="ios-close-circle"
                            />
                            支出
                          </Tab>
                        </Tabs>
                        <CategorySelect categories={filterCategories} 
                            selectedCategory={this.state.selectedCategory}
                            onSelectCategory={this.selectCategory}
                        />
                        <PriceForm 
                          onFormSubmit={this.submitForm}
                          onCancelSubmit={this.cancelSubmit}
                          item={editItem}
                        />
                    </div>
                </div>
            </div>
        );
    }
} 
 
export default withRouter(WithContext(Create));