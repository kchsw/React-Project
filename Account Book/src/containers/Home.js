import React, { Component } from 'react';
import logo from '../logo.svg';
import PriceList from '../components/PriceList'
import ViewTab from '../components/ViewTab'
import TotalPrice from '../components/TotalPrice'
import MonthPicker from '../components/MonthPicker'
import CreateBtn from '../components/CreateBtn'
import { LIST_VIEW, CHART_VIEW, TYPE_INCOME, TYPE_OUTCOME, parseDate, padLeft } from '../utility'

export  const categorys = {
    "1": {
            "id": 1,
            "name": "旅行",
            "type": "outcome",
            "iconName": "ios-plane"
    },
    "2": {
        "id": 2,
        "name": "理财",
        "type": "income",
        "iconName": "logo-yen"
    }

}
export const mockData = [
    {
      "id": 1,
      "title": "去云南旅行",
      "date": "2019/11/10",
      "price": 400,
      "cid": 1
    },
    {
      "id": 2,
      "title": "理财基金",
      "date": "2019/10/10",
      "price": 400,
      "cid": 2
    }
  ]

const newItem = {
        "id": 3,
        "title": "新添加项目",
        "date": "2019/11/10",
        "price": 600,
        "cid": 2
    }

    class Home extends Component {
        constructor(props){
            super(props)
            this.state = {  
                accountData: mockData,
                currentData: parseDate(),
                tabView: LIST_VIEW
            }
        }
        changeView = (view) => {
            this.setState({
                tabView: view
            })
        }
        changeDate = (year, month) => {
            this.setState({
                currentData: { year, month }
            })
        }
        modifyItem = (modifyiedItem) => {
            this.state.accountData.find(item => item.id === modifyiedItem.id).title = "nmsl"
            this.setState({
                accountData: this.state.accountData
            })
        }
        createItem = () => {
            this.setState({
                // accountData: mockData.concat(newItem)
                accountData: [newItem, ...this.state.accountData]
            })
        }
        deleteItem = (deletedItem) => {
            const filteredItems = this.state.accountData.filter(item => item.id !== deletedItem.id)
            this.setState({
                accountData: filteredItems
            })
        }
        render() { 
            let { accountData, currentData, tabView } = this.state
            accountData = accountData.map(item => {
                item.category = categorys[item.cid]
                return item
            }).filter(item => {
                return item.date.includes(`${currentData.year}/${padLeft(currentData.month)}`)
            })
            //这里直接对对象进行了操作
            // accountData.map(item => {
            //     item.category = categorys[item.cid]              
            // }) 
            let totalIncome = 0, totalOutcome = 0
            accountData.forEach(item => {
                item.category.type === TYPE_INCOME?
                totalIncome += item.price : totalOutcome += item.price
            })
            
            return (  
                <React.Fragment>
                    <header className="App-header">
                        <img src={logo} className="App-logo" alt="logo" />
                    </header> 
                    <div className="card" style={{width: '65%', margin: '0 auto'}}>
                        <div className="card-body">
                            <div className="row header">
                                <div className="col">
                                    <MonthPicker 
                                        year={currentData.year}
                                        month={currentData.month}
                                        onChange={this.changeDate}
                                    />
                                </div>
                                <div className="col d-flex justify-content-between align-items-center">
                                    <TotalPrice
                                        income={totalIncome}
                                        outcome={totalOutcome}
                                    />
                                </div>
                            </div>
                            <div className="content-area">
                                <ViewTab
                                    activeTab={tabView}
                                    onTabChange={this.changeView}
                                />
                                <CreateBtn onClick={this.createItem} />
                                {   tabView === LIST_VIEW &&
                                    <PriceList 
                                        items={accountData}
                                        onModifuItem={this.modifyItem}
                                        onDeleteItem={this.deleteItem}
                                    />
                                }
                                {
                                    tabView === CHART_VIEW &&
                                    <div>图表区域</div>
                                }
                            </div>
                        </div>
                    </div>
                </React.Fragment>
            );
        }
    }
   
  export default Home;