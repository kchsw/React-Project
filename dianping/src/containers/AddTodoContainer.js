import { connect } from 'react-redux'
import AddTodo from '../components/AddTodo'
import { addTodo, setTodoText } from '../store/actionCreators'
import { getText } from '../store/selectors'

const mapStateToProps = (state) => {
    return {
        text: getText(state)
    }
}
const mapDispatchToProps = (dispatch) => {
    return {
        setTodoText(text) {
            dispatch(setTodoText(text))
        },
        addTodo(text) {
            dispatch(addTodo(text))
        }
    }
}
 
export default connect(mapStateToProps, mapDispatchToProps)(AddTodo) 