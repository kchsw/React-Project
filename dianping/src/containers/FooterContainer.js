import { connect } from 'react-redux'
import Footer from '../components/Footer'
import { setFilter } from '../store/actionCreators'
import { getFilter } from '../store/selectors'

const mapStateToProps = (state) => {
    return {
        filter: getFilter(state)
    }
}
const mapDispatchToProps = (dispatch) => {
    return {
        setVisibleFillter(filter) {
            dispatch(setFilter(filter))
        }
    }
}
 
export default connect(mapStateToProps, mapDispatchToProps)(Footer) 