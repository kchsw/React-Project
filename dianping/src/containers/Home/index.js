import React, { Component } from 'react';
import Category from './components/Category'
import Headline from './components/Headline'
import Discount from './components/Discount'
import LikeList from './components/Likelist'
import HomeHeader from './components/HomeHeader'
import Banner from './components/Banner'
import Activity from './components/Activity'
import Footer from '../../components/Footer/index.js'

import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { actions as homeActions, getLikes, getDiscounts, getPageCountOfLikes } from '../../redux/modules/home'
class Home extends Component {
    state = {  }
    render() { 
        const { discounts, likes, pageCount } = this.props
        return ( 
            <>  
                <HomeHeader />
                <Banner dark />
                <Category />
                <Headline />
                <Activity />
                <Discount data={discounts}/>
                <LikeList 
                    data={likes} 
                    pageCount={pageCount}
                    fetchData = {this.fetchMoreLikes}
                />
                <Footer />
           </>
        );
    }
    componentDidMount() {
        this.props.homeActions.loadDiscounts()
    }
    fetchMoreLikes = () => {
        this.props.homeActions.loadLikes()
    }
}

const mapStateToProps = (state, props) => {
    return {
        likes: getLikes(state),
        discounts: getDiscounts(state),
        pageCount: getPageCountOfLikes(state)
    };
  };
  
const mapDispatchToProps = dispatch => {
    return {
        homeActions: bindActionCreators(homeActions, dispatch)
    };
};
 
export default connect(mapStateToProps, mapDispatchToProps)(Home);