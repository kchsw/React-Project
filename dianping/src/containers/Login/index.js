import React, { Component } from 'react';
import LoginHeader from './components/LoginHeader'
import LoginForm from './components/LoginForm'
import { Redirect } from 'react-router-dom'
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { 
    actions as loginActions, 
    getUsername,
    getPassword,
    isLogin
} from '../../redux/modules/login'

class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }
    render() { 
        const { username, password, isLogin, location: {state} } = this.props
        if(isLogin) {
            if(state && state.from) return <Redirect to={state.from} />
            return <Redirect to="/user" />
        }
        return (  
            <>
                <LoginHeader />
                <LoginForm 
                    username={username}
                    password={password}
                    isLogin={isLogin}
                    onChange={this.handleChange}
                    onSubmit={this.handleSubmit}
                />
            </>
        );
    }

    handleChange = (e) => {
        if(e.target.name === 'username') this.props.loginActions.setUsername(e.target.value)
        if(e.target.name === 'password') this.props.loginActions.setPassword(e.target.value)
    }
    handleSubmit = () =>{
        this.props.loginActions.login()
    }
}

const mapStateToProps = (state, props) => {
    return {
        username: getUsername(state),
        password: getPassword(state),
        isLogin: isLogin(state)
    };
};
  
const mapDispatchToProps = dispatch => {
    return {
        loginActions: bindActionCreators(loginActions, dispatch)
    };
};
 
export default connect(mapStateToProps, mapDispatchToProps)(Login);