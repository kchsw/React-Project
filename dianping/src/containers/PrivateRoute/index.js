import React from 'react';
import { Route, Redirect } from 'react-router-dom'
import { connect } from "react-redux";
import { 
    isLogin
} from '../../redux/modules/login'
const PrivateRoute = (props) => {
    const { component: Component, isLogin, ...rest } = props
    return (  
        <Route 
            {...rest}
            render={
                props => {
                    return isLogin ?
                        (<Component {...props}/>) :
                        (<Redirect to={{pathname: '/login', state: {from: props.location}}} />)
                }
            }
        />
    );
}

const mapStateToProps = (state, props) => {
    return {
        isLogin: isLogin(state)
    };
};
  
export default connect(mapStateToProps, null)(PrivateRoute);