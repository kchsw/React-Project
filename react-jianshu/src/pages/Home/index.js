import React, { Component } from 'react';
import Topic from './components/Topic'
import List from './components/List'
import Recommend from './components/Recommend'
import Writer from './components/Writer'
import { connect } from 'react-redux'
import { InitHomeList, toggleTopShow } from './store/actionCreators'
import {
    HomeWrapper,
    HomeLeft,
    HomeRight,
    BackTop
} from './style'
class Home extends Component {
    constructor(props) {
        super(props);
        this.state = {  }
    }
    handleScrollTop(){
        window.scroll(0, 0)
    }
    render() { 
        return (  
            <HomeWrapper>
                <HomeLeft>
                    <a href="/">
                    <img className="banner-img" src="https://upload.jianshu.io/admin_banners/web_images/4686/b205842c3dd6736c233b052411fe1211f164fcaf.png?imageMogr2/auto-orient/strip|imageView2/1/w/1250/h/540"/>
                    </a>
                    <Topic></Topic>
                    <List></List>
                </HomeLeft>
                <HomeRight>
                    <Recommend></Recommend>
                    <Writer></Writer>
                </HomeRight>
                {
                    this.props.showScroll ? <BackTop onClick={this.handleScrollTop}>回到顶部</BackTop> : null
                }
                
            </HomeWrapper>
        );
    }
    componentDidMount(){
        this.props.initHomeData()
        this.bindEvents()
    }

    bindEvents(){
        window.addEventListener('scroll', this.props.toggleScrollShow)
    }

    componentWillUnmount(){
        window.removeEventListener('scroll', this.props.toggleScrollShow)
    }
}

const mapState = (state) =>{
    return {
        showScroll: state.getIn(['home', 'showScroll'])
    }
}

const mapDispatch = (dispatch) => {
    return {
        initHomeData(){
            dispatch(InitHomeList())
        },
        toggleScrollShow(e){
            document.documentElement.scrollTop > 200 ? dispatch(toggleTopShow(true)) : dispatch(toggleTopShow(false))
        }
    }
}
export default connect(mapState, mapDispatch)(Home);