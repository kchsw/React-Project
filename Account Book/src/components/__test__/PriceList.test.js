import React from 'react';
import { shallow } from 'enzyme';
import PriceList from '../PriceList'
import Ionicon from 'react-ionicons'
import { mockData, categorys } from '../../containers/Home'

const accountData = mockData.map(item => {
    item.category = categorys[item.cid]
    return item
})

const props = {
    items: accountData,
    onModifuItem: jest.fn(),
    onDeleteItem: jest.fn()
}

let wrapper 

describe('test PriceList component', () => {
    beforeEach(() => {
        wrapper = shallow(<PriceList {...props}/>)
    })
    it('snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
    it('should render correct price item length', () => {
        expect(wrapper.find('.list-group-item').length).toEqual(accountData.length)
    })
    it('should render correct icon and price for each item', () => {
        const iconList = wrapper.find('.list-group-item').first().find(Ionicon)
        expect(iconList.length).toEqual(3)
        expect(iconList.first().props().icon)
        .toEqual(accountData[0].category.iconName)
    })
    it('should trigger the correct function callbacks', () => {
        const firstItem =  wrapper.find('.list-group-item').first()
        firstItem.find('a').first().simulate('click')
        expect(props.onModifuItem).toHaveBeenCalledWith(accountData[0])
        firstItem.find('a').last().simulate('click')
        expect(props.onDeleteItem).toHaveBeenCalledWith(accountData[0])
    })
})