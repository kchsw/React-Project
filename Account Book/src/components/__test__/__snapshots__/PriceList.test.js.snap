// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PriceList component snapshot 1`] = `ShallowWrapper {}`;
